client/chat/channel.tsx

```diff
@@ -427,23 +475,29 @@ class Channel extends React.Component {
   }
 }
 
-const mapStateToProps = state => {
+const mapStateToProps = (state: RootState) => {
   return {
     user: state.auth.user,
     chat: state.chat,
   }
 }
 
-function isLeavingChannel(oldProps, newProps) {
+interface ChatChannelViewProps {
+  user: SelfUserRecord
+  chat: typeof ChatState
+  dispatch: DispatchFunction<any> // TODO(2Pac): Type this better
```

Comment:
Yes, do this in this PR :) (and the one below it)

I assume you didn't type this how we have it in DispatchRegistry because we're dispatching old-style actions in here, but I'd prefer that you type this correctly and cast those to `any` when you dispatch them instead. That way we get some type safety now, and as we migrate those things we get ever-more safety.

As for params, it's just what comes in from wouter, so it's:
`{ channel: string }`